File: lt0073.cpp
#include <vector>
#include <cstring>

using namespace std;

class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        int row =  matrix.size();
        
        if(row == 0)
            return;
        int col = matrix[0].size();
        
        vector<int> rows;
        vector<int> cols;

        for(int i = 0; i < row; i++)
        {
            for(int j = 0; j < col; j++)
            {
                if(matrix[i][j] == 0)
                {
                    rows.push_back(i);
                    cols.push_back(j);
                }
            }
        }


        for(int i = 0; i < rows.size(); i++)
        {
            for(int j = 0; j < col; j++)
                matrix[rows[i]][j] = 0;
        }

        for(int i = 0; i < cols.size(); i++)
        {
            for(int j = 0; j < row; j++)
                matrix[rows[j]][i] = 0;
        }

        return;
              
    }
};

File: code/lt0073.java
// https://leetcode-cn.com/problems/set-matrix-zeroes/
// set matrix zeroes
class Solution {
    public void setZeroes(int[][] matrix) {
        int rows = matrix.length;
        if(rows == 0)
            return;
        int cols = matrix[0].length;
        if(cols == 0)
            return;
        
        boolean isCol = false;

        for(int i = 0; i < rows; i++){
            
            if(matrix[i][0] == 0)
                isCol = true;

            for(int j = 1; j < cols; j++){
                if(matrix[i][j] == 0){
                    matrix[i][0] = 0;
                    matrix[0][j] = 0;
                }
            }
        }



        for(int i = 1; i < rows; i++){
            for(int j = 1; j < cols; j++){
                if(matrix[i][0] == 0 || matrix[0][j] == 0)
                    matrix[i][j] = 0;
            }
        }

        if(matrix[0][0] == 0){
            for(int i = 1; i < cols; i++){
                matrix[0][i] = 0;
            }
        }

        if(isCol){
            for(int i = 0; i < rows; i++)
                matrix[i][0] = 0;
        }

        // if(matrix[0][0] == 0){
        //     for(int i = 0; i < rows; i++)
        //         matrix[i][0] = 0;
        //     for(int j = 0; j < cols; j++)
        //         matrix[0][j] = 0;
        // }

        return;
    }
}